.github/workflows/pull-request.yaml

```diff
@@ -1,13 +1,63 @@
-name: CI-pullrequest
+name: Pull request checks
 
 on: [pull_request]
 
 jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-go@v4
+        with:
+          go-version: '1.22'
```

Comment:
The code is currently using go1.19, I think we should use the same version here to avoid any possible problems